File: kotlin/lib/src/main/kotlin/models/BackgroundTaskFinishedEvent2.kt
// This file is @generated
package com.svix.kotlin.models

import com.svix.kotlin.StringAnyMapSerializer
import kotlinx.serialization.Serializable

@Serializable
data class BackgroundTaskFinishedEvent2(
    @Serializable(with = StringAnyMapSerializer::class) val data: Map<String, Any>,
    val status: BackgroundTaskStatus,
    val task: BackgroundTaskType,
    val taskId: String,
)

File: javascript/src/models/backgroundTaskFinishedEvent2.ts
// this file is @generated
/* eslint @typescript-eslint/no-explicit-any: 0 */
import {
  BackgroundTaskStatus,
  BackgroundTaskStatusSerializer,
} from "./backgroundTaskStatus";
import { BackgroundTaskType, BackgroundTaskTypeSerializer } from "./backgroundTaskType";

export interface BackgroundTaskFinishedEvent2 {
  data: any;
  status: BackgroundTaskStatus;
  task: BackgroundTaskType;
  /** The QueueBackgroundTask's ID. */
  taskId: string;
}

export const BackgroundTaskFinishedEvent2Serializer = {
  _fromJsonObject(object: any): BackgroundTaskFinishedEvent2 {
    return {
      data: object["data"],
      status: BackgroundTaskStatusSerializer._fromJsonObject(object["status"]),
      task: BackgroundTaskTypeSerializer._fromJsonObject(object["task"]),
      taskId: object["taskId"],
    };
  },

  _toJsonObject(self: BackgroundTaskFinishedEvent2): any {
    return {
      data: self.data,
      status: BackgroundTaskStatusSerializer._toJsonObject(self.status),
      task: BackgroundTaskTypeSerializer._toJsonObject(self.task),
      taskId: self.taskId,
    };
  },
};
